synapse/rest/client/user_directory.py

```diff
@@ -63,8 +63,8 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonMapping]:
 
         body = parse_json_object_from_request(request)
 
-        limit = body.get("limit", 10)
-        limit = min(limit, 50)
+        limit = int(body.get("limit", 10))
+        limit = max(min(limit, 50), 0)
```

Comment:
The call to `int` sneakily allows the limit to be given as a string. I can look the other way for a drive-by fix though.